### ISSUE TITLE
Blurry UI on Wayland + GNOME without fractional scaling

### ISSUE THREAD

**Issue by aolieman** [microsoft/vscode#203303]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.85.2
- OS Version: Ubuntu 23.10 (Linux 6.5.0-14-generic)

Steps to Reproduce:

1. Run a Linux distribution + Wayland + GNOME
2. Multi-display setup: 100% scaling on the primary monitor, 200% scaling on the secondary monitor
3. Run VS Code with `--enable-features=WaylandWindowDecorations --disable-features=WaylandFractionalScaleV1 --ozone-platform=wayland`
4. Observe that the UI is blurry on the 200% scaling display
5. Use the "reload window" command to render the VS Code window sharply

> Everything looks great and sharp when I have the screens positioned so that the one with fractional scaling is to the left or on top of the other screen. Fonts are blurry if the screens are positioned in the other two possible locations.

> Using the reload window command from the palette solves the issue in these cases.

_Originally posted by @v-tibi in https://github.com/microsoft/vscode/issues/192590#issuecomment-1735713448_

Even without fractional scaling enabled, this is still an issue on GNOME 45.1. I'm very happy to have learned about the reload window workaround. My multi-monitor setup is indeed similar, minus the fractional scaling: 100% display positioned above a 200% display. Having to reload the VS Code window regularly is still a hassle, however.

My non-exhaustive list of events that cause the window to become blurry, to be resolved with a manual reload:

- after moving it to another monitor
- after using the lock screen

**Comment by deepak1556** [comment]
Very likely related to https://bugs.chromium.org/p/chromium/issues/detail?id=1474176

**Comment by aolieman** [comment]
This seems unlikely because I'm not using a fractional scale for any of the displays in my setup. I've also disabled Chromium's feature toggle for fractional scaling.

I run other Electron apps that don't suffer from the same issue while using Wayland. They may be running in different Electron and Chromium versions.

**Comment by deepak1556** [comment]
Can you check with our insiders version https://code.visualstudio.com/insiders which is on a newer runtime version.

**Comment by halfmexican** [comment]
Running the insiders version appears to fix the issue

``code-insiders --enable-features=UseOzonePlatform --enable-features=WaylandWindowDecorations --ozone-platform=wayland --disable-features=WaylandFractionalScaleV1
``

This fixes blurry fonts on my 4k GNOME Desktop running at 200% scaling

**Comment by nocnoc94** [comment]
Here is how to get it to work with the normal version of vscode

First copy the desktop file

`cp /usr/share/applications/code.desktop ~/.local/share/applications/    `

Then add additional arguments to the desktop file

```
[Desktop Entry]
Name=Visual Studio Code
Comment=Code Editing. Redefined.
GenericName=Text Editor
# Exec=/usr/share/code/code --unity-launch %F
Exec=/usr/share/code/code --enable-features=UseOzonePlatform,WaylandWindowDecorations --ozone-platform-hint=auto --unity-launch %F
```

**Comment by MightyMirko** [comment]
> Here is how to get it to work with the normal version of vscode
> 
> First copy the desktop file
> 
> `cp /usr/share/applications/code.desktop ~/.local/share/applications/ `
> 
> Then add additional arguments to the desktop file
> 
> ```
> [Desktop Entry]
> Name=Visual Studio Code
> Comment=Code Editing. Redefined.
> GenericName=Text Editor
> # Exec=/usr/share/code/code --unity-launch %F
> Exec=/usr/share/code/code --enable-features=UseOzonePlatform,WaylandWindowDecorations --ozone-platform-hint=auto --unity-launch %F
> ```

After reading this i tried this on Ubuntu 22.04 Wayland. Unfortunately, it did not fix the issue. 

Calling code 

`code --enable-features=UseOzonePlatform --ozone-platform=wayland`
like this fixed it. 

I have an alias now for 
`code='code --enable-features=UseOzonePlatform --ozone-platform=wayland'`

**Comment by JohannBi** [comment]
Running the insiders version works well to me too (ubuntu 23.10)

```code-insiders --enable-features=UseOzonePlatform --enable-features=WaylandWindowDecorations --ozone-platform=wayland --disable-features=WaylandFractionalScaleV1```

However, is there a way to set it permanently ?

**Comment by halfmexican** [comment]
> Running the insiders version works well to me too (ubuntu 23.10)
> 
> `code-insiders --enable-features=UseOzonePlatform --enable-features=WaylandWindowDecorations --ozone-platform=wayland --disable-features=WaylandFractionalScaleV1`
> 
> However, is there a way to set it permanently ?

You can edit your desktop entry so it always runs

**Comment by JohannBi** [comment]
Thank you, it works !

**Comment by aolieman** [comment]
I can also confirm that the issue is resolved in the newer insiders build. This can be closed once the fix makes it into the stable release.

**Comment by rdev32** [comment]
> Here is how to get it to work with the normal version of vscode
> 
> First copy the desktop file
> 
> `cp /usr/share/applications/code.desktop ~/.local/share/applications/ `
> 
> Then add additional arguments to the desktop file
> 
> ```
> [Desktop Entry]
> Name=Visual Studio Code
> Comment=Code Editing. Redefined.
> GenericName=Text Editor
> # Exec=/usr/share/code/code --unity-launch %F
> Exec=/usr/share/code/code --enable-features=UseOzonePlatform,WaylandWindowDecorations --ozone-platform-hint=auto --unity-launch %F
> ```

this works on fedora 38+ with the stable build ty!

**Comment by aolieman** [comment]
Resolved in 1.88.0 on Ubuntu 23.10 as well.

**Comment by jhpratt** [comment]
I still need the flags to not have slightly blurry text for some reason, and I'm on the latest version. Fedora 40.

```text
Version: 1.89.1
Commit: dc96b837cf6bb4af9cd736aa3af08cf8279f7685
Date: 2024-05-07T05:16:23.416Z
Electron: 28.2.8
ElectronBuildId: 27744544
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Linux x64 6.8.11-300.fc40.x86_64
```
